### PR TITLE
GPS/GAL L1L5 configuration to use Skylark

### PR DESCRIPTION
**Pull request by tao-swift**
This PR creates a configuration file to perform GPS&GAL L1/L5 RTK using Skylark corrections. In case dual-frequency L1/L2 RTK is needed, the `pos1-frequency` needs to be changed to `l1+l2`, and the skylark correction source `inpstr2-path` needs to be changed to `na.skylark.swiftnav.com:2101/CRS`.

### CODE REVIEW

**Comment by dgburr** [comment]
What about pos1-wlmodear? I can't see any handling for it in https://github.com/swift-nav/RTKLIB/blob/master/src/options.c

File: app/rtkrcv/rtk_skylark_l1l5.conf [hunk 1]
```diff
@@ -0,0 +1,151 @@
+# RTKNAVI options (2016/09/19 07:28:53, v.2.4.3 b24)
+
+console-passwd     =admin
+console-timetype   =gpst       # (0:gpst,1:utc,2:jst,3:tow)
+console-soltype    =dms        # (0:dms,1:deg,2:xyz,3:enu,4:pyl)
+console-solflag    =1          # (0:off,1:std+2:age/ratio/ns)
+
+pos1-posmode       =kinematic  # (0:single,1:dgps,2:kinematic,3:static,4:movingbase,5:fixed,6:ppp-kine,7:ppp-static,8:ppp-fixed)
+pos1-frequency     =l1+l2+l5         # (1:l1,2:l1+l2,3:l1+l2+l5,4:l1+l5)
```
Comment (dgburr) [inline app/rtkrcv/rtk_skylark_l1l5.conf:9]: Add # note: change this to l1+l2 for L1/L2 receivers

File: app/rtkrcv/rtk_skylark_l1l5.conf [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+file-tempdir       =
+file-geexefile     =
+file-solstatfile   =
+file-tracefile     =
+
+inpstr1-type       =ntripcli   # (0:off,1:serial,2:file,3:tcpsvr,4:tcpcli,7:ntripcli,8:ftp,9:http)
+inpstr2-type       =ntripcli   # (0:off,1:serial,2:file,3:tcpsvr,4:tcpcli,7:ntripcli,8:ftp,9:http)
+inpstr3-type       =off        # (0:off,1:serial,2:file,3:tcpsvr,4:tcpcli,7:ntripcli,8:ftp,9:http)
+inpstr1-path       =username:password@igs-ip.net:2101/STFU00USA0
+inpstr2-path       =username:password@na2.skylark.swiftnav.com:2101/CRS
```
Comment (dgburr) [inline app/rtkrcv/rtk_skylark_l1l5.conf:124]: Add `note: host name may vary depending upon geographic region and frequency ranges`

**Comment by tao-swift** [comment]
> What about `pos1-wlmodear`? I can't see any handling for `prcopt.wlmodear` in https://github.com/swift-nav/RTKLIB/blob/master/src/options.c
> 
> (obviously we want to comment this out by default, but the option should be supported for customers who want to use a static rover)

Already done in https://github.com/swift-nav/RTKLIB/blob/ad3581fb62415541473a30f8079d3428072ee76f/src/options.c#L58

**Comment by dgburr** [comment]
> > What about `pos1-wlmodear`? I can't see any handling for `prcopt.wlmodear` in https://github.com/swift-nav/RTKLIB/blob/master/src/options.c
> > (obviously we want to comment this out by default, but the option should be supported for customers who want to use a static rover)
> 
> Already done in
> 
> https://github.com/swift-nav/RTKLIB/blob/ad3581fb62415541473a30f8079d3428072ee76f/src/options.c#L58

That is for `prcopt.modear`, not `prcopt.wlmodear`

**Comment by tao-swift** [comment]
> > > What about `pos1-wlmodear`? I can't see any handling for `prcopt.wlmodear` in https://github.com/swift-nav/RTKLIB/blob/master/src/options.c
> > > (obviously we want to comment this out by default, but the option should be supported for customers who want to use a static rover)
> > 
> > 
> > Already done in
> > https://github.com/swift-nav/RTKLIB/blob/ad3581fb62415541473a30f8079d3428072ee76f/src/options.c#L58
> 
> That is for `prcopt.modear`, not `prcopt.wlmodear`

I was under the impression that we didn't want to recommend the user to use fix and hold, so didn't consider that.

**Comment by dgburr** [comment]
> > > > What about `pos1-wlmodear`? I can't see any handling for `prcopt.wlmodear` in https://github.com/swift-nav/RTKLIB/blob/master/src/options.c
> > > > (obviously we want to comment this out by default, but the option should be supported for customers who want to use a static rover)
> > > 
> > > 
> > > Already done in
> > > https://github.com/swift-nav/RTKLIB/blob/ad3581fb62415541473a30f8079d3428072ee76f/src/options.c#L58
> > 
> > 
> > That is for `prcopt.modear`, not `prcopt.wlmodear`
> 
> I was under the impression that we didn't want to recommend the user to use fix and hold, so didn't consider that.

We recommend it to the user for static scenarios, but not for dynamic scenarios (see draft Application Note)

File: app/rtkrcv/rtk_skylark_l1l5.conf [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+inpstr3-path       =
+inpstr1-format     =rtcm3      # (0:rtcm2,1:rtcm3,2:oem4,3:oem3,4:ubx,5:ss2,6:hemis,7:skytraq,8:gw10,9:javad,10:nvs,11:binex,12:rt17,13:sbf,14:cmr,17:sp3)
+inpstr2-format     =rtcm3      # (0:rtcm2,1:rtcm3,2:oem4,3:oem3,4:ubx,5:ss2,6:hemis,7:skytraq,8:gw10,9:javad,10:nvs,11:binex,12:rt17,13:sbf,14:cmr,17:sp3)
+inpstr3-format     =rtcm3      # (0:rtcm2,1:rtcm3,2:oem4,3:oem3,4:ubx,5:ss2,6:hemis,7:skytraq,8:gw10,9:javad,10:nvs,11:binex,12:rt17,13:sbf,14:cmr,17:sp3)
+inpstr2-nmeareq    =latlon     # (0:off,1:latlon,2:single)
```
Comment (dgburr) [inline app/rtkrcv/rtk_skylark_l1l5.conf:130]: Suggest changing this to `single` then remove `inpstr2-nmealat` and `inpstr2-nmealon`

**Review by dgburr — CHANGES_REQUESTED** [review]

**Comment by tao-swift** [comment]
> > > > > What about `pos1-wlmodear`? I can't see any handling for `prcopt.wlmodear` in https://github.com/swift-nav/RTKLIB/blob/master/src/options.c
> > > > > (obviously we want to comment this out by default, but the option should be supported for customers who want to use a static rover)
> > > > 
> > > > 
> > > > Already done in
> > > > https://github.com/swift-nav/RTKLIB/blob/ad3581fb62415541473a30f8079d3428072ee76f/src/options.c#L58
> > > 
> > > 
> > > That is for `prcopt.modear`, not `prcopt.wlmodear`
> > 
> > 
> > I was under the impression that we didn't want to recommend the user to use fix and hold, so didn't consider that.
> 
> We recommend it to the user for static scenarios, but not for dynamic scenarios (see draft Application Note)

Done!

File: app/rtkrcv/rtk_skylark_l1l5.conf (hunk 3 above)
Comment (tao-swift) [reply app/rtkrcv/rtk_skylark_l1l5.conf:130]: Actually during my testing, when setting to `single` mode, there was no skylark observations coming in. Only when specifying the lat and lon, it worked.

File: app/rtkrcv/rtk_skylark_l1l5.conf [hunk 4]
```diff
@@ -26,10 +26,11 @@ pos1-posopt4       =on         # (0:off,1:on)
 pos1-posopt5       =on         # (0:off,1:on)
 pos1-posopt6       =off        # (0:off,1:on)
 pos1-exclsats      =C02        # (prn ...)
-pos1-navsys        =9         # (1:gps+2:sbas+4:glo+8:gal+16:qzs+32:comp)
-pos2-armode        =wide-lane #fix-and-hold # (0:off,1:continuous,2:instantaneous,3:fix-and-hold)
+pos1-navsys        =9          # (1:gps+2:sbas+4:glo+8:gal+16:qzs+32:comp)
+pos2-armode        =wide-lane  #fix-and-hold # (0:off,1:continuous,2:instantaneous,3:fix-and-hold)
+pos2-wlarmode      =off        # (0:off,1:on)
```
Comment (dgburr) [inline app/rtkrcv/rtk_skylark_l1l5.conf:31]: Add `note: only enable for static rover`

File: app/rtkrcv/rtk_skylark_l1l5.conf (hunk 3 above)
Comment (dgburr) [reply app/rtkrcv/rtk_skylark_l1l5.conf:130]: Hmmm.... I will try and reproduce the problem on the bench.  It'll be a problem for customers if they need to use a hard-coded lat/lon!

Comment (tao-swift) [reply app/rtkrcv/rtk_skylark_l1l5.conf]: Figured out the reason. There are some IGS or GA's stations only sending observation messages, in such a case, SPP doesn't have a solution, and then corrections can not be sent. Changed it to `single`, but should be noted that if no ephemeris provided on the user end, the user needs to specify the lat and lon.

File: app/rtkrcv/rtk_skylark_l1l5.conf [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+pos2-basesig       =0          # (m)
+out-solformat      =llh        # (0:llh,1:xyz,2:enu,3:nmea)
+out-outhead        =off        # (0:off,1:on)
+out-outopt         =off        # (0:off,1:on)
+out-timesys        =gpst        # (0:gpst,1:utc,2:jst)
+out-timeform       =tow        # (0:tow,1:hms)
+out-timendec       =3
+out-degform        =dms        # (0:deg,1:dms)
+out-fieldsep       =
+out-outsingle      =off        # (0:off,1:on)
```
Comment (dgburr) [inline app/rtkrcv/rtk_skylark_l1l5.conf:61]: Set to `on` to match RINEX examples in Application Note

**Review by dgburr — APPROVED** [review]
LGTM

**Pull request merged**